Connectors/CSi/Speckle.Connectors.CSiShared/Bindings/CsiSharedSelectionBinding.cs

```diff
@@ -39,7 +44,7 @@ private void CheckSelectionChanged()
     if (!_lastSelection.SetEquals(currentIds))
     {
       _lastSelection = currentIds;
-      Parent.Send(SelectionBindingEvents.SET_SELECTION, currentSelection);
+      _threadContext.RunOnMain(() => Parent.Send(SelectionBindingEvents.SET_SELECTION, currentSelection));
```

Comment:
This is an unawaited Task now. Need to use an exception handler

Reply:
see updated commit

Reply:
Good catch! I've added the missing field for the top-level exception handler and wrapped the `Parent.Send` call with `_topLevelExceptionHandler.CatchUnhandled`